Remember search text in Find/Replace in Files

miv391 opened the pull request:
Fixes https://github.com/godotengine/godot-proposals/issues/6340

Find/Replace in Files dialog works now using the same operating logic as vscode. If the user has selected something in the code editor, that selection is used. But if nothing is selected, then previous search string is not cleared. In Replace mode focus is set either to Find text edit or Replace text edit depending on what the user is likely to type next.

- Find in Files 
	- some text is selected -> use that as Find string 
	- no text is selected -> do not change Find string

- Replace in Files 
	- some text is selected -> use that as Find string, focus Replace text edit 
	- no text is selected -> do not change Find string, focus Find text edit

aaronfranke committed:
Move 2D-only resources to their own folder

aaronfranke committed:
Move 3D-only resources to their own folder

Calinou committed:
Use black for font outlines by default instead of white

This makes font outlines more usable out of the box, as black
is one of the most commonly used colors for font outlines.

Calinou committed:
Display the build date in the editor and when starting the engine

This can be used to quickly see how recent a development build is,
without having to look up the commit date manually.
When juggling around with various builds (e.g. for benchmarking),
this can also be used to ensure that you're actually running the
binary you intended to run.

The date stored is the date of the Git commit that is built, not
the current date at the time of building the binary. This ensures
binaries can remain reproducible.

The version timestamp can be accessed using the `timestamp` key
of the `Engine.get_version_info()` return value.

Mickeon committed:
Tweak SceneTree Documentation

Mickeon committed:
Overhaul Engine documentation

akien-mga committed:
astcenc: Update to 4.7.0

https://github.com/ARM-software/astc-encoder/releases/tag/4.5.0
https://github.com/ARM-software/astc-encoder/releases/tag/4.6.0
https://github.com/ARM-software/astc-encoder/releases/tag/4.7.0

smix8 committed:
Change ShapeCast3D and RayCast3D debug to use RenderingServer directly

Changes ShapeCast3D and RayCast3D debug to use RenderingServer directly.

Calinou committed:
Document Bounce = 1.0 not being sufficient for infinite energy conservation

aaronfranke committed:
Move 2D-only navigation code to its own subfolder

aaronfranke committed:
Move 3D-only navigation code to its own subfolder

miv391 committed:
Remember search text in Find/Replace in Files

AThousandShips commented on editor/find_in_files.cpp:
```diff
@@ -393,8 +393,24 @@ FindInFilesDialog::FindInFilesDialog() {
-	_search_text_line_edit->set_text(text);
-	_on_search_text_modified(text);
+	if (_mode == SEARCH_MODE) {
+		if (text != "") {
```
```suggestion
		if (!text.is_empty()) {
```
Same below

miv391 replied:
Fixed.

smix8 committed:
Auto-bake edited NavigationPolygons in the Editor on a timer

Auto-bakes edited NavigationPolygons in the Editor on a timer.

bitsawer committed:
Fix Volumetric Fog VoxelGI updates

YeldhamDev committed:
Add option to add built-in strings in the POT generation

Mickeon committed:
Optimise Object's `get_argument_options`

Mickeon committed:
Add autocompletion for ProjectSettings' methods

Mickeon committed:
Add autocompletion for InputMap's methods

Mickeon committed:
Add autocompletion for ClassDB & AudioServer

akx committed:
Replace hand-written pre-commit hooks with `pre-commit` Python tool

`pre-commit` can be installed with pip, and configured in the Godot repo with
`pre-commit install`. It can then easily be run both locally with
`pre-commit run`, and on CI, in a cross-platform way.

This makes it much easier for contributors to set up pre-commit hooks,
without having to manually copy files to their git folder.

Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>

Mickeon committed:
Overhaul Quaternion documentation

Mickeon committed:
Add autocompletion for TranslationServer

Mickeon committed:
Optimize SceneTree's `change_scene_to_file` autocompletion

Mickeon committed:
Replace built-in docs codeblock's leading spaces with tabs

Replace built-in docs codeblock's leading spaces with tabs

AThousandShips committed:
Prevent threading problems in `TileMap`

Calinou committed:
Document using RichTextLabel's `meta_clicked` to handle clickable URLs

davthedev committed:
Fix item positioning & pointer detection areas of ItemList

shana committed:
VS: Improve performance of parsing project file

VS doesn't handle string parsing very well, so having all the files in one
property slows down VS a lot when loading the projects. Splitting the files
up into per-directory properties brings down project processing times from
20 seconds to 1 second (on my machine).

Muller-Castro committed:
Add const lvalue ref to editor/* container parameters

dsnopek committed:
WebXR: Add support for hand tracking

fire committed:
Add ufbx for FBX importing

This update introduces a new import method for FBX files using ufbx. If the fbx2gltf import fails, it will use the most recently cached scene from the ufbx import. The process is sped up by introducing threads to load the ufbx portion.

Key changes include:

- Support for importing geometry helper nodes in FBX files.
- Addition of cameras and lights with updated names.
- Removal of the fbx importer manager.
- Introduction of ModelDocument3D and updates to its methods.
- Changes to FBX import options and visibility.
- Updating the documentation and handling some errors.
- Store the original non-unique node, mesh and animation names in FBX and glTF.

Co-Authored-By: bqqbarbhg <bqqbarbhg@gmail.com>

RandomShaper committed:
Direct3D 12: Enhance management of texture data life cycle

RandomShaper committed:
Direct3D 12: Implement proper fallback for format casting

RandomShaper committed:
Improve thread safety of resource loading

paulloz committed:
Cleanup C# projects, code quality & style

New rules:
- Do not silence CA1805 any more
- Limit where we silence CA1707, CA1711, CA1720
- Enforce severity=warning for IDE0040
- Enforce Allman style braces
- Enforce naming conventions (IDE1006 is still severity=suggestion)

Fixes:
- Fix REFL045, CS1572, CS1573
- Suppress CS0618 when generating `InvokeGodotClassMethod`
- Fix indent when generating GD_constants.cs
- Temporarily silence CS1734 in generated code
- Fix a lot of naming rule violations

Misc.:
- Remove ReSharper comments for RedundantNameQualifier
- Remove suppression attributes for RedundantNameQualifier
- Remove severity=warnings for CA1716, CA1304 (already included in the level of analysis we run)

RandomShaper committed:
Enhance cache modes in resource loading

- Unify documentation, hoping to clear misconcepctions about about propagation of the cache mode across dependant loads.
- Clarify in docs that `CACHE_MODE_REPLACE` now also works on the main resource (from #87008).
- Add two recursive modes, counterparts of `CACHE_MODE_REPLACE` and `CACHE_MODE_IGNORE`, since it seems some need them (see #59669, #82830).
- Let resources, even loaded with one of the ignore-cache modes, get a path, which is useful for tools.

deralmas committed:
Wayland: Restore tablet support and handle multiple tools

This code was already partially there, although heavily incomplete and
nowadays commented out.

It got broken after the `WaylandThread` refactor and I didn't bother to
bring it over, preferring to `#if 0` it into oblivion for the time
being as I don't have a tablet/pen which support an eraser and tilt
reporting.

This commit brings it back and adds proper multi-tool support (needed
for eraser detection) thanks to winston-yallow, who could test this code
with their more capable tablet.

clayjohn committed:
Properly calculate binormal when creating SurfaceTool from arrays

ryevdokimov committed:
Remove word duplicates in comments and strings, and fix casing and punctuation

akien-mga committed:
Merge pull request #88726 from RandomShaper/node_cust_leak

Fix leak of scene used for customization during export

akien-mga committed:
Merge pull request #88731 from ajreckof/Removing-element-in-array-will-not-change-page-when-emptying-page-

Fix removing element in array inspector not changing page when emptying page.

clayjohn committed:
Avoid singularity when generated tangents and validate that tangents are good enough when using compression

akien-mga committed:
Merge pull request #81746 from fire/vsk-ufbx

Add ufbx for FBX importing

akien-mga committed:
Merge pull request #88411 from dsnopek/webxr-hand-tracking

WebXR: Add support for hand tracking

akien-mga committed:
Merge pull request #88736 from dsnopek/ci-godot-cpp-42

Use godot-cpp 4.2 for the "Godot CPP" CI workflow

dsnopek committed:
Fix lint errors in WebXR module

akien-mga committed:
Merge pull request #88740 from dsnopek/webxr-linting-issues

Fix lint errors in WebXR module

MajorMcDoom committed:
Fixed the text editor theme not being applied on editor start.

clayjohn committed:
Consistently use system_fbo instead of binding 0 as it is needed for iOS devices

BastiaanOlij committed:
OpenXR: Register eye gaze even when disabled

stoofin committed:
Add all Viewport.debug_draw enum options to dropdown

AThousandShips committed:
[Doc] Fix GDScript casing of `String.num_scientific`

ryevdokimov committed:
Show instantiating message with position when using to drag to instantiate in 3D

KoBeWi committed:
Fix inefficient list iteration in Node

MajorMcDoom committed:
Make Skeleton3D::add_bone return the new bone index right away, instead of requiring an additional call to get_bone_count.

Malcolmnixon committed:
Add body tracking support.

nongvantinh committed:
Fix Script Editor saves C# files as embedded scripts

fixes: #88543
fixes: #88160

Because of the limitations of compiled programming languages like C#, when a newly created *.cs file hasn't been compiled, we don't have any information about its `Path` or `Type` in the `assemply`. This means we end up creating an invalid instance of this file whenever there's a request. Consequently, multiple instances of the script can exist. When a new instance takes over the path, it clears the `path_cache` of the previous instance, leading to undefined behavior.

ryevdokimov committed:
If spinbox text can't be parsed use last updated text

lyuma committed:
Add new scene import option to import as Skeleton

Adds a bool import option `nodes/import_as_skeleton_bones`.
This is supported in all FBX or GLTF document based formats.
It is especially useful for retargeting and importing animations.

lyuma committed:
Add rest fixer toggle to disable bone pose reset

Adds a "Reset All Bone Poses After Import" option to the Skeleton3D Rest Fixer settings.
Default value of true resets the bone poses to rest (usually a t-pose), matching previous behavior.
If disabled, keeps the bones posed as they would have been before silhouette adjustment.

akien-mga committed:
Merge pull request #85501 from /remove-packed-scene-cache

akien-mga committed:
Merge pull request #85905 from bruvzg/win_app_id

[Windows] Set application user model ID to prevent editor / running project and different versions of editor taskbar icon stacking.

akien-mga committed:
Merge pull request #87181 from Mickeon/doc-peeves-quaternion-i-keep-misspelling-it

Overhaul Quaternion documentation

akien-mga committed:
Merge pull request #88694 from bruvzg/str_checks

[macOS] Add null checks for `NSString stringWithUTF8String`.

akien-mga committed:
Merge pull request #88733 from ryevdokimov/Remove-word-duplicates

Remove word duplicates in comments and strings, and fix casing and punctuation

akien-mga committed:
Merge pull request #88767 from AThousandShips/string_doc_fix

[Doc] Fix GDScript casing of `String.num_scientific`

akien-mga committed:
Merge pull request #88785 from KoBeWi/list_cringe

Fix inefficient list iteration in Node

Sauermann committed:
Document that `parse_input_event` doesn't influence the OS

Sauermann committed:
Mention how Control focus loss affects `Viewport.gui_focus_change`

RedMser committed:
Inspector changes indicator propagates upwards

Mickeon committed:
Fix toggling a plugin makes its name black

jsjtxietian committed:
Do overflow check in when gererating NoiseTexture3D

adamscott committed:
Add create folder icon

mhilbrunner committed:
Fix removing connected peer during disconnection

ryevdokimov committed:
Show instantiating or inherit message when using to drag to instantiate in 2D

emre0altan committed:
Save scene and global checkbox options of History window, fixes #88432

MajorMcDoom committed:
Add necessary elaboration to documentation for Node3D::get_parent_node_3d

ajreckof committed:
Fix paste Value can empty a dictionary depending on right-click location

smix8 committed:
Split monolithic physics class files

Splits monolithic physics class files.

dalexeev committed:
GDScript: Canonicalize script path in FQCN

AThousandShips committed:
[Animation] Improvements to `Tween` memory management

Using `ObjectID` instead of manually breaking reference cycles.

akien-mga committed:
Merge pull request #83691 from theraot/slidedocs

Expanded documentation for vector slide and project

akien-mga committed:
Merge pull request #88589 from RandomShaper/gl_preview_liveunlock

Rework viewport capture in preview generation

akien-mga committed:
Merge pull request #88611 from Alex2782/force_device_cpu_architecture

Display a warning if device CPU architecture is not active in the export preset.

akien-mga committed:
Merge pull request #88615 from Rindbee/add-h_separation-between-icons-in-CheckButton-and-CheckBox

Add a `h_separation` between icons in `CheckButton`/`CheckBox`

akien-mga committed:
Merge pull request #88675 from MajorMcDoom/script-editor-redundancy

Remove some redundant method calls from ScriptEditor

akien-mga committed:
Merge pull request #88692 from nongvantinh/fix-88543

Fix Script Editor saves C# files as embedded scripts.

akien-mga committed:
Merge pull request #88730 from raulsntos/dotnet/deprecated-message

C#: Add deprecated message to generated bindings

akien-mga committed:
Merge pull request #88754 from BastiaanOlij/fix_openxr_eye_gaze_init

OpenXR: Register eye gaze even when disabled

akien-mga committed:
Merge pull request #88755 from stoofin/debug-draw-enum-names

Add all `Viewport.debug_draw` enum options to dropdown

akien-mga committed:
Merge pull request #88791 from MajorMcDoom/add-bone-index

Make `Skeleton3D::add_bone` return the new bone index

akien-mga committed:
Merge pull request #88811 from Sauermann/fix-focus-change-doc

Mention how Control focus loss affects `Viewport.gui_focus_change`

akien-mga committed:
Merge pull request #88814 from RedMser/inherit-property-revert-inspector

Inspector `(N changes)` indicator propagates upwards

akien-mga committed:
Merge pull request #88826 from mhilbrunner/net-fix-has-erase

Networking scene multiplayer: Fix removing connected peer during disconnection

akien-mga committed:
Merge pull request #88853 from dalexeev/gds-canonicalize-fqcn

GDScript: Canonicalize script path in FQCN

akien-mga committed:
Merge pull request #83133 from aaronfranke/2d-navigation

Move 2D-only and 3D-only navigation code to its own subfolder

akien-mga committed:
Merge pull request #88849 from ajreckof/Fix-paste-Value-can-empty-a-dictionary-depending-on-right-click-location

Fix paste Value can empty a dictionary depending on right-click location

akien-mga committed:
Merge pull request #50148 from aaronfranke/move-resources

Move 2D and 3D resources to their own folders

akien-mga committed:
Don't attempt shutting down adb on exit if not started

Fixes #88864.

dsnopek committed:
GDExtension: Fix calling ancestor methods on runtime classes

clayjohn committed:
Add compositor_free branch in Compatibility scene renderer free function

aaronfranke committed:
Explain gutter types in the TextEdit documentation

dalexeev committed:
GDScript: Fix type highlighting after whitespace

Sauermann committed:
Fix owner when importing meshes

Handle the case correctly, that a child of a replaced node has
as owner an ancestor of the replaced node.

clayjohn committed:
Include tangent in SurfaceTool vertex compare operator

smix8 committed:
Fix NavigationObstacle2D debug being affected by Node2D transform

Fixes NavigationObstacle2D debug being affected by Node2D transform.

akien-mga committed:
Merge pull request #68649 from Mickeon/doc-peeves-sweet-tree-scenery

Tweak SceneTree Documentation

akien-mga committed:
Merge pull request #80372 from Mickeon/doc-peeves-engineering

Overhaul Engine singleton documentation

akien-mga committed:
Merge pull request #82968 from Calinou/doc-physics-bounce-damping

Document Bounce = 1.0 not being sufficient for infinite energy conservation

akien-mga committed:
Merge pull request #87973 from flooxo/master

Remove duplicate `ARRAY_CUSTOM_RGBA8_UNORM` ref in Mesh docs

akien-mga committed:
Merge pull request #88368 from Muller-Castro/value2ref-editor

Add const lvalue ref to `editor/*` container parameters

akien-mga committed:
Merge pull request #88574 from Malcolmnixon/openxr_1_0_34

OpenXR: Update to version 1.0.34

akien-mga committed:
Merge pull request #88575 from davthedev/itemlist-no-separators

Hide separators in ItemList in editor theme

akien-mga committed:
Merge pull request #88652 from RobProductions/fix-hover-margin

Fix editor hover style margins when Draw Extra Borders is enabled

akien-mga committed:
Merge pull request #88725 from clayjohn/GLTF-binormal-sign

Properly calculate binormal when creating SurfaceTool from arrays

akien-mga committed:
Merge pull request #88738 from clayjohn/mesh_compression-tangents

Multiple fixes for compressed meshes

akien-mga committed:
Merge pull request #88742 from MajorMcDoom/text-editor-zoom

Fix the text editor theme not being applied on editor start

akien-mga committed:
Merge pull request #88745 from clayjohn/GLES3-system-fbo

Consistently use `system_fbo` instead of binding 0 as it is needed for iOS devices

akien-mga committed:
Merge pull request #88774 from ryevdokimov/instantiating-message

Show instantiating message with position when using to drag to instantiate in 3D

akien-mga committed:
Merge pull request #88819 from lyuma/import_nodes_as_skeleton_bones

Add new scene import option to import as Skeleton

akien-mga committed:
Merge pull request #88821 from lyuma/preserve_initial_pose

Allow preserving the initial bone pose in rest fixer

akien-mga committed:
Merge pull request #88823 from jsjtxietian/limit-NoiseTexture3D

Do overflow check in when generating NoiseTexture3D

akien-mga committed:
Merge pull request #88825 from adamscott/add-create-folder-icon

Add create folder icon

akien-mga committed:
Merge pull request #88829 from ryevdokimov/institate_message-2d

Show instantiating or inherit message when using drag to instantiate in 2D

akien-mga committed:
Merge pull request #88856 from AThousandShips/tween_fix

[Animation] Improvements to `Tween` memory management

akien-mga committed:
Merge pull request #88865 from akien-mga/adb-ywz

Don't attempt shutting down adb on exit if not started

akien-mga committed:
Merge pull request #88866 from akien-mga/pre-commit

Replace hand-written pre-commit hooks with `pre-commit` Python tool

akien-mga committed:
Merge pull request #88870 from dsnopek/gdextension-fix-runtime-class-ancestor-methods

GDExtension: Fix calling ancestor methods on runtime classes

akien-mga committed:
Merge pull request #88873 from clayjohn/GLES3-compositor-free

Add `compositor_free` branch in Compatibility scene renderer free function

akien-mga committed:
Merge pull request #88875 from aaronfranke/doc-textedit-gutter-type

Explain gutter types in the TextEdit documentation

akien-mga committed:
Merge pull request #88876 from dalexeev/gds-fix-type-highlighting-after-whitespace

GDScript: Fix type highlighting after whitespace

akien-mga committed:
Merge pull request #88890 from clayjohn/SurfaceTool-tangent-hash

Include tangent in SurfaceTool vertex compare operator

akien-mga committed:
Merge pull request #88895 from smix8/navobstacle2d_debug_transform

Fix NavigationObstacle2D debug being affected by Node2D transform

akien-mga committed:
EditorFileSystem: Add verbose print for file being (re)imported

Also print the time it took for each file.

emre0altan committed:
Clarify C# Color opacity comment

Chaosus committed:
Remove redundant space after enum/flags word in editor docs

AThousandShips committed:
[TileSet] Fix crash when deleting dragged polygon point

mihe committed:
Fix crash when selecting re-added `Tree` item

akien-mga committed:
Merge pull request #87478 from AThousandShips/tile_thread_fix_2

Prevent threading problems in `TileMap`

akien-mga committed:
Merge pull request #87872 from RandomShaper/d3d12_tex_mem_is_life

Direct3D 12: Enhance management of texture data life cycle

akien-mga committed:
Merge pull request #88817 from Mickeon/plugin-list-black

Fix toggling a plugin makes its name black

akien-mga committed:
Merge pull request #88862 from smix8/split_physics_classes

Split monolithic physics class files

akien-mga committed:
Merge pull request #88912 from AThousandShips/tile_drag_fix

[TileSet] Fix crash when deleting dragged polygon point

blackbird806 committed:
Remove self includes in some files

AThousandShips committed:
[Doc] Fix some incorrect uses of "children"

Repiteo committed:
C#: Only check for empty deprecation message if deprecated

akien-mga committed:
Merge pull request #88027 from RandomShaper/d3d12_sdfgi_cool

Direct 3D 12: Implement proper fallback for format casting

akien-mga committed:
Merge pull request #88918 from blackbird806/patch-1

Remove self includes in some files

akien-mga committed:
Merge pull request #88921 from Repiteo/dotnet/deprecate-check-fix

C#: Only check for empty deprecation message if deprecated

clayjohn committed:
Fix wrong indexing when generating dummy tangents in GLTF import

raulsntos committed:
Move dotnet-format script to pre-commit

Co-authored-by: Paul Joannon <437025+paulloz@users.noreply.github.com>

bruvzg committed:
[macOS/Windows] Fix application indicator destruction.

akien-mga committed:
Merge pull request #59247 from Calinou/editor-display-build-date

Display the build date in the editor and when starting the engine

akien-mga committed:
Merge pull request #82122 from dalexeev/gds-add-export-hidden-annotation

GDScript: Add `@export_storage` annotation

akien-mga committed:
Merge pull request #82599 from smix8/shapecast_debug_4.x

Change ShapeCast3D and RayCast3D debug to use RenderingServer directly

akien-mga committed:
Merge pull request #87504 from smix8/nav_mesh_auto_rebake

Auto-bake edited NavigationPolygons in the Editor on a timer

akien-mga committed:
Merge pull request #88497 from Sauermann/fix-owner-import

Fix owner when importing meshes

akien-mga committed:
Merge pull request #88570 from paulloz/dotnet/warnings-editorconfig-cleanup

[.NET] Better `.editorconfig` setup in `modules/mono/`

akien-mga committed:
Merge pull request #88909 from emre0altan/clarify-c#-color-opacity-comment

Clarify C# Color opacity comment

akien-mga committed:
Merge pull request #88911 from Chaosus/fix_editor_help

Remove redundant space after enum/flags word in editor docs

akien-mga committed:
Merge pull request #88931 from clayjohn/GLTF-normals

Fix wrong indexing when generating dummy tangents in GLTF import

akien-mga committed:
Merge pull request #88932 from bruvzg/fix_indicatore_destroy

[macOS/Windows] Fix application indicator destruction.

akien-mga committed:
Merge pull request #88917 from mihe/tree-selection-crash

Fix crash when selecting re-added `TreeItem::Cell`

clayjohn committed:
Ensure proper vertex input masks are used in 2D compatibility renderer

ajreckof committed:
Fix crash on changing resource type of dictionary new key/value while editing the resource

akien-mga committed:
Merge pull request #88938 from clayjohn/GLES3-skeleton2d

Ensure proper vertex input masks are used in 2D compatibility renderer

lawnjelly committed:
Fix physics platform crash

Physics body previously stored the RID of a collision object and accessed it on the next frame, leading to a crash if the object had been deleted.
This PR checks the object still exists via the ObjectID prior to access.

akien-mga committed:
Merge pull request #88572 from ajreckof/fix-crash-on-dictionarry-new-key/value-change-type-when-inspector-opened

Fix crash on changing resource type of dictionary new key/value while editing the resource.

akien-mga committed:
Merge pull request #88794 from ryevdokimov/fix-spinbox

If SpinBox text can't be parsed use last updated text

akien-mga committed:
Merge pull request #88798 from Malcolmnixon/xr-body-tracker

XR: Add body tracking support.

dalexeev committed:
Core: Add `Callable.create` static method for `Variant` callables

akien-mga committed:
RTL: Clarify that line, paragraph, and character numbers are zero-indexed

Closes #82885.

398utubzyt committed:
Fix compilation errors when `DISABLE_DEPRECATED` is defined

akien-mga committed:
Pre-commit: Update to clang-format 17.0.6 and black 24.2.0

AThousandShips committed:
[Doc] Clarify some details about deferred calls

adamscott committed:
Hide snap_2d_vertices_to_pixel setting to advanced

groud committed:
Remove almost all remaining dependencies of TileMapLayer on TileMap

KoBeWi committed:
Improve cursor_set_custom_image() method

bruvzg committed:
[macOS] Enable input from controllers in the background.

Calinou committed:
Add percent (`%`) sign to Remainder node name in visual shaders

This makes the node easier to search in the Create New Node dialog.

akien-mga committed:
Fix linking NavigationServer2D/3D with all classes disabled in build profile

For some reason the destructor for NavigationMeshSourceGeometryData2D/3D
being implemented in the .cpp was causing linking issues.

bruvzg committed:
[macOS] Fix some deprecation warnings.

Sauermann committed:
Fix mouse entered notifications

Make sure, that a windows initial state is `mouse_in_viewport = false`.
This makes sure, that the mouse entered notification is sent when the
mouse hovers a window for the first time.

For embedded Windows, `NOTIFICATION_WM_MOUSE_ENTER` is currently sent
twice in a row. Remove one of the places where it is sent.
`Window::_update_mouse_over()` is the correct one, because there it is
also called for native windows.

jsjtxietian committed:
Change shader compiler default setting to avoid doctool error

akien-mga committed:
Don't access Node3D/Node2D/Control global transform in `reparent` unless needed

Fixes #89002.

Co-authored-by: Tamás Gálffy <ezittgtx@gmail.com>

joaoh82 committed:
Add RandomNumberGenerator::rand_weighted method

akien-mga committed:
Merge pull request #54641 from Calinou/gui-tweak-default-outline-color

Use black for font outlines by default instead of white

akien-mga committed:
Merge pull request #81822 from nlupugla/nodepath-slice

Add `NodePath::slice` method

akien-mga committed:
Merge pull request #86222 from YeldhamDev/give_me_those_strings_baby

Add option to add built-in strings in the POT generation

akien-mga committed:
Merge pull request #86705 from KoBeWi/resourception

Improve saving of built-in resources

akien-mga committed:
Merge pull request #87268 from Wyxaldir/master

Fix for `resource_local_to_scene` in arrays.

akien-mga committed:
Merge pull request #88255 from Calinou/doc-richtextlabel-meta-clicked

Document using RichTextLabel's `meta_clicked` to handle clickable URLs

akien-mga committed:
Merge pull request #88561 from RandomShaper/res_load_safer

Improve thread safety of resource loading

akien-mga committed:
Merge pull request #88577 from davthedev/itemlist-light-refactor

Fix item positioning, text alignment & unwanted clipping of ItemList items

akien-mga committed:
Merge pull request #88664 from RandomShaper/res_load_modes

Enhance cache modes in resource loading

akien-mga committed:
Merge pull request #88744 from Riteo/wayland-multitool-tablet

Wayland: Restore tablet support and handle multiple tools

akien-mga committed:
Merge pull request #88904 from akien-mga/efs-tell-me-what-you-import

EditorFileSystem: Add verbose print for file being (re)imported

akien-mga committed:
Merge pull request #88920 from AThousandShips/group_doc_fix

[Doc] Fix some incorrect uses of "children"

akien-mga committed:
Merge pull request #88948 from dalexeev/core-add-callable-create

Core: Add `Callable.create` static method for `Variant` callables

akien-mga committed:
Merge pull request #88949 from shana/vs-performance-improvement

Visual Studio: Improve performance of parsing project file

akien-mga committed:
Merge pull request #88956 from akien-mga/doc-rtl-zero-idx

RTL: Clarify that line, paragraph, and character numbers are zero-indexed

akien-mga committed:
Merge pull request #88958 from 398utubzyt/gltf/use-correct-name

Fix compilation errors when `DISABLE_DEPRECATED` is defined

akien-mga committed:
Merge pull request #88959 from akien-mga/pre-commit-clang-17.0.6-black-24.2.0

Pre-commit: Update to clang-format 17.0.6 and black 24.2.0

akien-mga committed:
Merge pull request #88961 from AThousandShips/deferred_doc

[Doc] Clarify some details about deferred calls

akien-mga committed:
Merge pull request #88963 from adamscott/hide-to-advanced-snap-2d-vertices

Move `rendering/2d/snap/snap_2d_vertices_to_pixel` setting to advanced view

akien-mga committed:
Merge pull request #88966 from groud/remove_tilemap_layer_dependencies_to_tilemap

Remove almost all remaining dependencies of TileMapLayer on TileMap

akien-mga committed:
Merge pull request #88970 from KoBeWi/ImageCursor2D

Improve `cursor_set_custom_image()` method

akien-mga committed:
Merge pull request #88978 from bruvzg/macos_bg_cont

[macOS] Enable input from controllers in the background.

akien-mga committed:
Merge pull request #88980 from Calinou/visualshader-remainder-add-percent-sign

Add percent (`%`) sign to Remainder node name in visual shaders

akien-mga committed:
Merge pull request #88986 from akien-mga/navigation-fix-blank-build-profile

Fix linking NavigationServer2D/3D with all classes disabled in build profile

akien-mga committed:
Merge pull request #88987 from bruvzg/macos_12_depr_warn

[macOS] Fix some deprecation warnings.

akien-mga committed:
Merge pull request #88992 from Sauermann/fix-window-mouse-notification

Fix mouse entered notifications

akien-mga committed:
Merge pull request #88996 from jsjtxietian/fix-doctool

Change shader compiler default setting to avoid doctool error

dalexeev committed:
GDScript: Fix paths in `ResourceFormatLoaderGDScript::load()`

TCROC committed:
Fix C# epsilon compiler error on double precision build

akien-mga committed:
Merge pull request #88994 from TCROC/fix-c#-missing-epsilon-64bit

Fix C# epsilon compiler error on double precision build

akien-mga committed:
Merge pull request #89003 from akien-mga/node3d-reparent-no-global-transform

Don't access Node3D/Node2D/Control global transform in `reparent` unless needed

akien-mga committed:
Merge pull request #89005 from dalexeev/gds-fix-res-format-loader-path

GDScript: Fix paths in `ResourceFormatLoaderGDScript::load()`

smix8 committed:
Fix NavigationObstacle3D debug not reacting to visiblity changes

Fixes NavigationObstacle3D debug not reacting to visiblity changes.

bruvzg committed:
[RTL] Add optional `push_meta` argument to control how meta underline is drawn.

Mickeon committed:
Overhaul some "uncommon" wording in class reference

YeldhamDev committed:
Add auto translation to `Tree`

wheatear-dev committed:
Add unit tests for CryptoKey

akien-mga committed:
Merge pull request #80375 from akien-mga/astcenc-4.5.0

astcenc: Update to 4.7.0

akien-mga committed:
Merge pull request #85801 from HolonProduction/select-whole-words-line-edit

Allow dragging selection when selecting whole words in `LineEdit`

akien-mga committed:
Merge pull request #86023 from bitsawer/fix_fog_voxelgi

Fix Volumetric Fog VoxelGI updates

akien-mga committed:
Merge pull request #86398 from reach-satori/prof-visibility-fix

Fix internal profiling button being visible when disabled in settings

akien-mga committed:
Merge pull request #86743 from Mickeon/autocompletion-optimise-object

Optimise comparisons for Object's `get_argument_options`

akien-mga committed:
Merge pull request #86777 from Mickeon/autocomplete-classdb

Add autocompletion for ClassDB & AudioServer

akien-mga committed:
Merge pull request #87191 from Mickeon/autocompletion-TranslationServer

Add autocompletion for TranslationServer

Mickeon committed:
Add autocompletion for EditorSettings' methods

akien-mga committed:
Merge pull request #87301 from Mickeon/documentation-leading-spaces-to-tabulation

Replace built-in docs codeblock's leading spaces with tabs

akien-mga committed:
Merge pull request #88810 from Sauermann/fix-parse-input-doc

Document that `parse_input_event` doesn't influence the OS

akien-mga committed:
Merge pull request #88833 from emre0altan/save-scene-and-global-checkbox-options-in-history-window

Save scene and global checkbox options of History window

akien-mga committed:
Merge pull request #88841 from MajorMcDoom/doc-fix

Add necessary elaboration to documentation for `Node3D::get_parent_node_3d`

akien-mga committed:
Merge pull request #88883 from joaoh82/add-weighted-random-method

Add `RandomNumberGenerator::rand_weighted` method

akien-mga committed:
Merge pull request #88947 from lawnjelly/fix_physics_platform_rid

Fix physics platform crash

akien-mga committed:
Merge pull request #89009 from smix8/navobstacle_debug_visibility

Fix NavigationObstacle3D debug not reacting to visiblity changes

akien-mga committed:
Merge pull request #89024 from bruvzg/rtl_hover_meta

[RTL] Add optional `push_meta` argument to control how meta underline is drawn.

KoBeWi committed:
Fix update_property() in particle material editor

akien-mga committed:
Merge pull request #86758 from Mickeon/autocompletion-project-settings

Add autocompletion for ProjectSettings' methods

akien-mga committed:
Merge pull request #86764 from Mickeon/autocompletion-inputmap

Add autocompletion for InputMap's methods

akien-mga committed:
Merge pull request #87197 from Mickeon/autocompletion-SceneTree-change-packed-to-file

Optimize SceneTree's `change_scene_to_file` autocompletion

akien-mga committed:
Merge pull request #89021 from wheatear-dev/test-crypto-key

Add unit tests for CryptoKey

akien-mga committed:
Merge pull request #89032 from YeldhamDev/as_arvores_somos_nozes

Add auto translation to `Tree`

akien-mga committed:
Merge pull request #89036 from Mickeon/documentation-uncommon-words-2

Overhaul some "uncommon" wording in class reference

akien-mga committed:
Merge pull request #89043 from Mickeon/autocomplete-EditorSettings

Add autocompletion for EditorSettings' methods

akien-mga committed:
Merge pull request #89048 from KoBeWi/iBug

Fix `update_property()` in particle material editor

akien-mga committed:
Merge pull request #88933 from raulsntos/dotnet/pre-commit

Move dotnet-format script to pre-commit

miv391 committed:
Merge branch 'remember-search-text' of https://github.com/miv391/godot into remember-search-text

# Conflicts:
#	editor/find_in_files.cpp
